actions/get_single_sites.php

```diff
@@ -0,0 +1,162 @@
+$output = '';
+
+if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+
+    if (isset($_POST['page'])) {
+        $page = $_POST["page"];  
+    } else {
+        $page = 1;  
+    }
+
+    if (isset($_POST['state'])) {
+        $state = $_POST['state'];
+    } else {
+        $state = "all";
+    }
+} else  {  
+    $page = 1;  
+}  
+
+
+$offset = ($page - 1) * $records_per_page;
+$result = $migrationDAO->getSingleSitesByState($LINK->id, $state, $offset, $records_per_page);
+
+$output .= "<table class='table flex-wrap' id='singlesites_tbl'><tbody>";  
+
+foreach($result as $row) {
+    $output .= '<tr>  
+                <td><span>'.$row["started_at"].'</span></td> ';
+                
+    if($row['state'] == "completed") {
+        $output .= ' <td colspan="2"><span class="label alert-success"><strong>State: </strong>'.$row["state"].'</span></td>';
+    } else if($row['state'] == "init" || $row['state'] == "starting" || $row['state'] == "running" || $row['state'] == "importing" || $row['state'] == "exporting") {
+        $output .= ' <td colspan="2"><span class="label alert-info"><strong>State: </strong>'.$row["state"].'</span></td>';
+    } else if($row['state'] == "updating") {
+        $output .= ' <td colspan="2"><span class="label alert-info"><strong>State: </strong>'.$row["state"].'</span></td>';
+    } else if($row['state'] == "error") {
+        $output .= ' <td colspan="2"><span class="label alert-danger"><strong>State: </strong>'.$row["state"].'</span></td>';
+    }
+    
+    $output .= '<td colspan="2"><span>'.$row["title"].'</span></td>
+    <td colspan="2"><a href="https://vula.uct.ac.za/portal/site/'.$row["site_id"].'" target="_blank" class="img">
+        <img src="./static/img/vula.svg" class="img-fluid"/>
+        </a>
+    </td>';
+
+    if($row["imported_site_id"] > 0) {
+        $output .= '<td colspan="2"><a href="https://amathuba.uct.ac.za/d2l/home/'.$row["imported_site_id"].'" target="_blank" class="img">
+                        <img src="./static/img/amathuba_woodmark.svg" class="img-fluid"/>
+                        </a>
+                    </td>';
+    } else {
+        $output .= '<td colspan="2">&nbsp;</td>';
+    }
+
+    if($row["report_url"] != '' || $row["report_url"] != NULL) {
+        $output .= '<td colspan="2"><i class="fas fa-file-alt fa-2x text-primary show_report" data-toggle="modal" rel='.$row["report_url"].' data-target="#reportModal" id='.$row["site_id"].'></i></td>';
+    } else {
+        $output .= '<td colspan="2">&nbsp;</td>';
+    }
+    $output .= '</tr>';  
+}
+ 
+$output .= '</tbody></table>'; 
+
+$page_result = $migrationDAO->getAllSingleSitesByState($LINK->id, $state);
+$total_records = count($page_result);
+$total_pages = ceil($total_records/$records_per_page);  
+$previous_page = $page - 1;
+$next_page = $page + 1;
+$adjacents = "2";
+$second_last = $total_pages - 1;
+
+$output .='<div class="container text-center"><div class="btn-group flex-wrap" role="group">';
+
+if($page > 1) {
```

Comment:
This looks to be a bit excessive for displaying the page numbers ...

Maybe simplify it with something like:
```
<nav aria-label="Page navigation">
<ul class="pagination">
<?php if($page_no > 1){
echo "<li><a href='?page_no=1'>First Page</a></li>";
} ?>
    
<li <?php if($page_no <= 1){ echo "class='disabled'"; } ?>>
<a <?php if($page_no > 1){
echo "href='?page_no=$previous_page'";
} ?>>Previous</a>
</li>
    
<li <?php if($page_no >= $total_no_of_pages){
echo "class='disabled'";
} ?>>
<a <?php if($page_no < $total_no_of_pages) {
echo "href='?page_no=$next_page'";
} ?>>Next</a>
</li>

<?php if($page_no < $total_no_of_pages){
echo "<li><a href='?page_no=$total_no_of_pages'>Last &rsaquo;&rsaquo;</a></li>";
} ?>
</ul>
</nav>
```

Reply:
Done.